src/test/java/pl/coderslab/imageManager/mock/MockDatabase.java

```diff
@@ -0,0 +1,164 @@
+package pl.coderslab.imageManager.mock;
+
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import pl.coderslab.imageManager.image.Image;
+import pl.coderslab.imageManager.image.ImageRepository;
+
+import java.util.*;
+
+public class MockDatabase implements ImageRepository {
+    private static final Map<Long, Image> images = new HashMap<>();
+
+    @Override
+    public Image getByName(String name) {
+        Set<Map.Entry<Long, Image>> entrySet = images.entrySet();
+        Iterator<Map.Entry<Long, Image>> iterator = entrySet.iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<Long, Image> next = iterator.next();
+            String imageName = next
+                    .getValue()
+                    .getName();
+            if (imageName.equals(name)) {
+                return next.getValue();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public List<Image> findAll() {
+        List<Image> imagesList = new ArrayList<>();
+        Set<Map.Entry<Long, Image>> entrySet = images.entrySet();
```

Comment:
weź spróbuj w stylu images.entrySet().stream().Collect(Collectors.toList()), nie jestem pewien czy pyknie bo piszę z pamięci. Najwyżej zerknij na to, bo teraz jest na iteratorach = może być lepiej bo jest zbyt niskopoziomowo

https://mkyong.com/java8/java-8-convert-map-to-list/